cmake/modules/FindLibCrypto.cmake

```diff
@@ -15,83 +15,93 @@
 #  LibCrypto_SHARED_LIBRARY    The path to libcrypto.so
 #  LibCrypto_STATIC_LIBRARY    The path to libcrypto.a
 
-find_path(LibCrypto_INCLUDE_DIR
-    NAMES openssl/crypto.h
-    HINTS
-        ${CMAKE_PREFIX_PATH}/include
-        ${CMAKE_INSTALL_PREFIX}/include
-    )
-find_library(LibCrypto_SHARED_LIBRARY
-    NAMES libcrypto.so libcrypto.dylib
-    HINTS
-    ${CMAKE_PREFIX_PATH}/build/crypto
-    ${CMAKE_PREFIX_PATH}/build
-    ${CMAKE_PREFIX_PATH}
-    ${CMAKE_PREFIX_PATH}/lib64
-    ${CMAKE_PREFIX_PATH}/lib
-    ${CMAKE_INSTALL_PREFIX}/build/crypto
-    ${CMAKE_INSTALL_PREFIX}/build
-    ${CMAKE_INSTALL_PREFIX}
-    ${CMAKE_INSTALL_PREFIX}/lib64
-    ${CMAKE_INSTALL_PREFIX}/lib
-    )
-find_library(LibCrypto_STATIC_LIBRARY
-    NAMES libcrypto.a
-    HINTS
-    ${CMAKE_PREFIX_PATH}/build/crypto
-    ${CMAKE_PREFIX_PATH}/build
-    ${CMAKE_PREFIX_PATH}
-    ${CMAKE_PREFIX_PATH}/lib64
-    ${CMAKE_PREFIX_PATH}/lib
-    ${CMAKE_INSTALL_PREFIX}/build/crypto
-    ${CMAKE_INSTALL_PREFIX}/build
-    ${CMAKE_INSTALL_PREFIX}
-    ${CMAKE_INSTALL_PREFIX}/lib64
-    ${CMAKE_INSTALL_PREFIX}/lib
-    )
+find_package(crypto QUIET)
 
-if (BUILD_SHARED_LIBS)
-    set(LibCrypto_LIBRARY ${LibCrypto_SHARED_LIBRARY})
+if (crypto_FOUND)
+    get_target_property(crypto_INCLUDE_DIR crypto INTERFACE_INCLUDE_DIRECTORIES)
+    message(STATUS "aws-c-cal found target: crypto")
+    message(STATUS "crypto Include Dir: ${crypto_INCLUDE_DIR}")
+    set(LIBCRYPTO_FOUND true)
+    set(LibCrypto_FOUND true)
```

Comment:
I would put the comment here of why this is needed. I saw it in another PR, but otherwise I wouldn't know about it.

Reply:
Sure.